File: LeetCode 75 Sheet3/2024-06-19 20.28.38 Eslam Khder's Personal Meeting Room - Copy/c++/1768. Merge Strings Alternately.cpp
#include<iostream>
using namespace std;
                      
string mergeAlternately(string word1, string word2) {
    int sizeWord1 = word1.size();  // 1
    int sizeWord2 = word2.size(); // 3
    
    string result = ""; // "af"
    
    int w1 = 0;
    int w2 = 0;
    //     0         012
    //     a         fnm
    while(w1 < sizeWord1 || w2 < sizeWord2){
    	// w1 = 1     w2 = 1
    	if (w1 < sizeWord1){
    		result.push_back(word1[w1]);
    		w1++;
		}
		
		if (w2 < sizeWord2){
    		result.push_back(word2[w2]);
    		w2++;
		}
	}
	
	return result;
}
int main(){
	
}

File: LeetCode 75 Sheet3/2024-06-19 20.28.38 Eslam Khder's Personal Meeting Room - Copy/java/1768. Merge Strings Alternately.java
import java.util.Scanner;

public class Main {
    public static void main(String[] argv)
    {
        Scanner scanner = new Scanner(System.in);

    }


    public String mergeAlternately(String word1, String word2) {
        int sizeWord1 = word1.length(); // 1
        int sizeWord2 = word2.length(); // 3

        int w1 = 0; // 0 1
        int w2 = 0; // 0 1 2 3

        String result = ""; // "aert"

        //         0               012
        // word1= "a"      word2= "ert"
        while (w1 < sizeWord1 || w2 < sizeWord2) {

            if (w1 < sizeWord1) {
                result += word1.charAt(w1);
                w1++;
            }

            if (w2 < sizeWord2) {
                result += word2.charAt(w2);
                w2++;
            }
        }

        return result;
    }

}
